alpha/apps/kaltura/lib/kUploadTokenMgr.php

```diff
@@ -126,9 +126,34 @@ public function uploadFileToToken($fileData, $resume = false, $resumeAt = -1)
 		$allowedStatuses = array(UploadToken::UPLOAD_TOKEN_PENDING, UploadToken::UPLOAD_TOKEN_PARTIAL_UPLOAD);
 		if (!in_array($this->_uploadToken->getStatus(), $allowedStatuses, true))
 			throw new kUploadTokenException("Invalid upload token status", kUploadTokenException::UPLOAD_TOKEN_INVALID_STATUS);
-
+		
 		$this->updateFileName($fileData);
 		
+		$entryUpdated = false;
+		$entry = $this->getAttachedEntry();
+		if ($entry && $entry->getType() == entryType::MEDIA_CLIP && !$entry->getMediaType())
+		{
+			$mediaType = $this->getMediaType();
+			if ($mediaType)
+			{
+				$entry->setMediaType($this->getMediaType());
+				$entryUpdated = true;
+			}
+			
+			if($entry->getMediaType() == KalturaMediaType::IMAGE)
```

Comment:
you can use $mediaType var you already initialized and check
if($mediaType && $mediaType ==..)